### PR TITLE
CoW: Enable CoW by default and remove warning build

### PR DESCRIPTION
**Pull request by phofl**
I'd prefer to go this way in enabling CoW. Having it on makes it easier to enforce the change step by step rather than doing it all at once.

### CODE REVIEW

**Comment by phofl** [comment]
cc @mroeschke 

this should be green now and ready to merge hopefully. That allows us to move forward with the deprecations here

File: pandas/tests/test_downstream.py [hunk 1]
```diff
@@ -46,6 +46,7 @@ def test_dask(df):
         pd.set_option("compute.use_numexpr", olduse)
 
 
+@pytest.mark.xfail(reason="array is read-only")
```
Comment (mroeschke) [inline pandas/tests/test_downstream.py:49]: Can this test eventually work with cow?

File: pandas/tests/series/test_ufunc.py [hunk 2]
```diff
@@ -456,8 +454,7 @@ def add3(x, y, z):
         ufunc(ser, ser, df)
 
 
-# TODO(CoW) see https://github.com/pandas-dev/pandas/pull/51082
-@td.skip_copy_on_write_not_yet_implemented
+@pytest.mark.xfail(reason="see https://github.com/pandas-dev/pandas/pull/51082")
```
Comment (mroeschke) [inline pandas/tests/series/test_ufunc.py:457]: Is there a way to still test this with cow?

Comment (phofl) [reply pandas/tests/series/test_ufunc.py:457]: no not at the moment

File: pandas/tests/test_downstream.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/test_downstream.py]: yeah we can use literally any other function that doesn't modify inplace, changed to log

File: pandas/tests/series/test_ufunc.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/series/test_ufunc.py:457]: Okay sounds good.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Just a merge conflict post ArrayManager merge. Feel free to merge after that

**Pull request merged**